nixos/modules/image/repart.nix

```diff
@@ -60,13 +60,63 @@ let
       };
     };
   };
+
 in
 {
   options.image.repart = {
 
     name = lib.mkOption {
       type = lib.types.str;
-      description = lib.mdDoc "The name of the image.";
+      description = lib.mdDoc ''
+        Name of the image.
+
+        If this option is unset but config.system.image.id is set,
+        config.system.image.id is used as the default value.
+      '';
+    };
+
+    version = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
+      default = config.system.image.version;
```

Comment:
I think, this could result in invalid `IMAGE_VERSION` values as `system.image.version` accepts arbitrary string while `IMAGE_VERSION` does not.